aiidalab_widgets_base/viewers.py

```diff
@@ -1136,7 +1136,14 @@ def _observe_supercell(self, _=None):
             self.set_trait(
                 "displayed_structure", None
             )  # To make sure the structure is always updated.
-            self.set_trait("displayed_structure", self.structure.repeat(self.supercell))
+            # because nglview loads the structure through save and load a temp "pdb" file
+            # thus change the cell and positions into standard form.
+            # we need to mimic this process and update the cell and positions
+            pdb_structure = self.structure.copy()
+            pdb_structure.set_cell(
+                self.structure.cell.standard_form()[0], scale_atoms=True
+            )
+            self.set_trait("displayed_structure", pdb_structure.repeat(self.supercell))
```

Comment:
```suggestion
            # nglview converts the ASE structure internally
            # by saving and loading from temporary "pdb" file.
            # In this process, the cell and positions are converted to the standard form.
            # Because we use the `displayed_structure` trait outside of nglview
            # (e.g. to calculate and display custom bonds),
            # we need to do the same conversion here as well.
            oriented_structure = self.structure.copy()
            oriented_structure.set_cell(
                self.structure.cell.standard_form()[0], scale_atoms=True
            )
            self.set_trait("displayed_structure", oriented_structure.repeat(self.supercell))
```

Reply:
dear @superstar54 @danielhollas I tested and to me seems working.
I am a little bit scared of converting back and forth from pdb in case coordinates could lose accuracy but the tests I did seem to have no accuracy problems.

I wanted to update the files of the examples but I will do it on a separate PR as soon as I manage